src/Cardano/Wallet/Api/Types.hs

```diff
@@ -352,7 +325,15 @@ instance ToJSON ApiBlockData where
     toJSON = genericToJSON defaultRecordTypeOptions
 
 instance FromJSON ApiCoins where
-    parseJSON = genericParseJSON defaultRecordTypeOptions
+    parseJSON bytes = do
+        v@(ApiCoins _ (Quantity c)) <-
+            genericParseJSON defaultRecordTypeOptions bytes
+        if isValidCoin (Coin $ fromIntegral c)
+            then return v
+            else fail $
+                "invalid coin value: value has to be lower than"
```

Comment:
"lower than or equal"

Reply:
Fun fact: In English, "lower than" (or similarly, "higher than") have the same meaning as `<` and, we therefore require "lower than or equal" to mean `<=`.

In French, the literal translation for "lower than" would be "inférieur à", which in Maths, is interpreted as `<=`, and French would say "strictement inférieur à" to mean `<=` (which would translate to "strictly lower than"). 

Anyway, Good catch :sweat_smile:

Reply:
Interesting!

For this example, Croatian follows the same natural language rules as English does (probably that's the reason why it was easier to spot it for me - and easier to do the mistake for you). Interesting how we tend to think/solve-problems in our natural language constrains